pkg/gateway/controller.go

```diff
@@ -207,12 +212,34 @@ func getGatewayStatus(obj *unstructured.Unstructured) (haStatus string, connecti
 	return haStatus, connections, true
 }
 
+func getLocalClusterId(obj *unstructured.Unstructured) (clusterId string, found bool) {
+	status, found, err := unstructured.NestedMap(obj.Object, "status")
+	if !found || err != nil {
+		klog.Errorf("status field not found in %#v, err was: %v", obj, err)
+		return "", false
+	}
+
+	localEndpoint, found, err := unstructured.NestedMap(status, "localEndpoint")
+	if !found || err != nil {
+		klog.Errorf("localEndpoint field not found in %#v, err was: %v", status, err)
+		return "", false
+	}
+
+	clusterId, found, err = unstructured.NestedString(localEndpoint, "cluster_id")
+	if !found || err != nil {
+		klog.Errorf("cluster_id field not found in %#v, err was: %v", localEndpoint, err)
+		return "", false
+	}
+
+	return clusterId, true
+}
+
 func (c *Controller) getClusterStatusMap() map[string]bool {
 	return c.clusterStatusMap.Load().(map[string]bool)
 }
 
 func (c *Controller) IsConnected(clusterId string) bool {
-	return !c.gatewayAvailable || c.getClusterStatusMap()[clusterId]
+	return !c.gatewayAvailable || c.localCluster == clusterId || c.getClusterStatusMap()[clusterId]
```

Comment:
This is an optimization to avoid reading localEndpoint everytime, which will never change but can become nil if there are no gateways. This also helps with scenario where we can become the only cluster coz all others are gone. In such scenarios local services that have been exported should still be available. Ideally I'd like to remove reading `localEndpoint` altogether. `clusterid` is deployment config that is available multiple places, so should get at init somehow.

Reply:
You're obtaining the local clusterID from the `localEndpoint`  in the `Gateway`. But you're storing it as a field which can be accessed by multiple threads so technically needs to be synchronized. I'm suggesting to avoid this as outlined above.  Either way you're still obtaining the clusterID from the `localEndpoint`  in the `Gateway`.

Reply:
Done